simpa_tests/automatic_tests/test_bandpass_filter.py

```diff
@@ -15,55 +14,152 @@
+                plt.subplot(5, 1, i + 1)
+                plt.title(labels[i])
+                plt.plot(self.t_values, time_series)
+            plt.tight_layout()
+            plt.show()
 
-        filtered_time_series_with_settings = bandpass_filtering_with_settings(combined_time_series, settings,
-                                                                              settings[Tags.RECONSTRUCTION_MODEL_SETTINGS],
-                                                                              device)
+    def test_tukey_bandpass_filter(self):
+        filtered_time_series_with_settings = tukey_bandpass_filtering_with_settings(self.combined_time_series, self.settings,
+                                                                                    self.settings[Tags.RECONSTRUCTION_MODEL_SETTINGS],
+                                                                                    self.device)
 
-        filtered_time_series_with_resampling = bandpass_filtering(combined_time_series, 1e-3, int(11000), int(9000), 0, True)
+        filtered_time_series_with_resampling = tukey_bandpass_filtering(self.combined_time_series, 1e-3, int(11000), int(9000), 0, True)
 
-        filtered_time_series = bandpass_filtering(combined_time_series, 1e-3, int(11000), int(9000), 0)
+        filtered_time_series = tukey_bandpass_filtering(self.combined_time_series, 1e-3, int(11000), int(9000), 0)
 
         # check if both bandpass filtering methods return the same result
         assert np.array_equal(filtered_time_series, filtered_time_series_with_settings)
 
-        assert (np.abs(base_time_series - filtered_time_series) < 1e-5).all()
+        assert (np.abs(self.base_time_series - filtered_time_series) < 1e-5).all()
 
-        assert (np.abs(base_time_series - filtered_time_series_with_resampling) < 1e-1).all()
+        assert (np.abs(self.base_time_series - filtered_time_series_with_resampling) < 1e-1).all()
 
         if Visualize:
             labels = ["Base time series", "Low frequency time series", "High frequency time series",
                       "Combined time series", "Filtered time series"]
-            for i, time_series in enumerate([base_time_series, low_freq_time_series, high_freq_time_series,
-                                             combined_time_series, filtered_time_series]):
+            for i, time_series in enumerate([self.base_time_series, self.low_freq_time_series, self.high_freq_time_series, self.combined_time_series, filtered_time_series]):
                 plt.subplot(5, 1, i + 1)
                 plt.title(labels[i])
-                plt.plot(t_values, time_series)
+                plt.plot(self.t_values, time_series)
             plt.tight_layout()
             plt.show()
+
+    def visualize_filtered_spectrum(self, filtered_spectrum: np.ndarray):
+        if Visualize:
```

Comment:
I think it is more elegant to do the `if Visualize` within the tests and only call this function if `Visualize is True` in the first place.